src/Database/Kioku/Internal/TrieIndex.hs

```diff
@@ -81,7 +96,7 @@ lookupSubtrie key (TI buf rootDrop root) =
                                 <|> trySubtries rest
 
       in case (remainingKey, remainingArc) of
-         ("", _) -> Just $ TI buf (BS.length commonArc) offset
+         ("", _) -> Just $ TI buf (BS.length commonArc + arcDrop) offset
```

Comment:
:heart: 

Incidentally, it appears we've never caught this because we've never actually made use of anything that that uses the output of `lookupSubtrie` to then do further subtrie lookups. Only `firstStopAlong` currently does that, and it appears to not be used anywhere in `flipstone` or `exfreight-zeta`

Reply:
Yeah, we discovered that and added a task to our story to clean up the code by removing firstStopAlong.